bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessPolicies/UpdateAccessPolicyCommand.cs

```diff
@@ -8,24 +10,135 @@ namespace Bit.Commercial.Core.SecretsManager.Commands.AccessPolicies;
 public class UpdateAccessPolicyCommand : IUpdateAccessPolicyCommand
 {
     private readonly IAccessPolicyRepository _accessPolicyRepository;
+    private readonly ICurrentContext _currentContext;
+    private readonly IProjectRepository _projectRepository;
+    private readonly IServiceAccountRepository _serviceAccountRepository;
 
-    public UpdateAccessPolicyCommand(IAccessPolicyRepository accessPolicyRepository)
+    public UpdateAccessPolicyCommand(
+        IAccessPolicyRepository accessPolicyRepository,
+        ICurrentContext currentContext,
+        IProjectRepository projectRepository,
+        IServiceAccountRepository serviceAccountRepository)
     {
         _accessPolicyRepository = accessPolicyRepository;
+        _currentContext = currentContext;
+        _projectRepository = projectRepository;
+        _serviceAccountRepository = serviceAccountRepository;
     }
 
-    public async Task<BaseAccessPolicy> UpdateAsync(Guid id, bool read, bool write)
+    public async Task<BaseAccessPolicy> UpdateAsync(Guid id, bool read, bool write, Guid userId)
     {
         var accessPolicy = await _accessPolicyRepository.GetByIdAsync(id);
         if (accessPolicy == null)
         {
             throw new NotFoundException();
         }
 
+        switch (accessPolicy)
+        {
+            case UserProjectAccessPolicy ap:
+                if (ap.GrantedProjectId == null)
+                {
+                    throw new NotFoundException();
+                }
+
+                var project = await _projectRepository.GetByIdAsync(ap.GrantedProjectId.Value);
```

Comment:
Same comment as Delete, if we include the related objects directly we can simplify this quite a bit.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/cd92ea867e0c629cdebeccbb8a3d9499053cef04